ui/src/app/tasks/task-create/task-create.component.spec.ts

```diff
@@ -0,0 +1,162 @@
+  const commonTestParams = { id: 'faketask' };
+  const commonTestAppInfos = { faketask: {
+    name: 'fakename',
+    options: [
+      {id: 'fakename.opt1', name: 'opt1', defaultValue: 'val1'}
+    ]}};
+
+
+  beforeEach(async(() => {
+    activeRoute = new MockActivatedRoute();
+    TestBed.configureTestingModule({
+      declarations: [
+        TaskCreateComponent
+      ],
+      imports: [
+        BusyModule,
+        FormsModule,
+        ReactiveFormsModule,
+        RouterTestingModule.withRoutes([])
+      ],
+      providers: [
+        { provide: TasksService, useValue: tasksService },
+        { provide: ActivatedRoute, useValue: activeRoute },
+        { provide: ToastyService, useValue: toastyService }
+      ]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TaskCreateComponent);
+    component = fixture.componentInstance;
+  });
+
+  function updateDefinitionName(definitionName: string) {
+    component.definitionName.setValue(definitionName);
+  }
+
+  it('task create component should be created', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should have matching app title with given param id', () => {
+    activeRoute.testParams = commonTestParams;
+    tasksService.testAppInfos = commonTestAppInfos;
+    de = fixture.debugElement.query(By.css('h1'));
+    el = de.nativeElement;
+    fixture.detectChanges();
+
+    expect(el.textContent).toContain('Create Definition for Task App faketask');
+    expect(toastyService.testSuccess.length).toBe(1);
+    expect(toastyService.testSuccess).toContain('App info loaded.');
+  });
+
+  it('should warn about empty initial definition name', () => {
+    activeRoute.testParams = commonTestParams;
+    tasksService.testAppInfos = commonTestAppInfos;
+    de = fixture.debugElement.query(By.css('[id=commonDefinitionParameters] > div'));
+    el = de.nativeElement;
+
+    fixture.detectChanges();
+    expect(el.classList.contains('has-warning')).toBe(true);
+    expect(el.classList.contains('has-feedback')).toBe(true);
+  });
+
+  it('should warn about empty illegal definition name', () => {
+    activeRoute.testParams = commonTestParams;
+    tasksService.testAppInfos = commonTestAppInfos;
+    de = fixture.debugElement.query(By.css('[id=commonDefinitionParameters] > div'));
+    el = de.nativeElement;
+
+    updateDefinitionName('fake task');
+    fixture.detectChanges();
+    expect(el.classList.contains('has-warning')).toBe(true);
+    expect(el.classList.contains('has-feedback')).toBe(true);
+  });
+
+  it('should warn about same definition name', () => {
+    activeRoute.testParams = commonTestParams;
+    tasksService.testAppInfos = commonTestAppInfos;
+    de = fixture.debugElement.query(By.css('[id=commonDefinitionParameters] > div'));
+    el = de.nativeElement;
+
+    updateDefinitionName('validtaskname');
+    fixture.detectChanges();
+    expect(el.classList.contains('has-warning')).toBe(false);
```

Comment:
need to check that message is not show when valid.   The message is still present when definition name is valid, as discussed here:  https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/342.